PWGCF/EbyEFluctuations/Tasks/kaonIsospinFluctuations.cxx

```diff
@@ -1163,244 +1528,398 @@ struct KaonIsospinFluctuations {
+  {
+    float centrality = -1;
+    for (const auto& collision : collisions) {
+      if (!collision.has_mcCollision()) {
+        continue;
+      }
+      centrality = -1;
+      const auto& mcColl = collision.mcCollision();
+
+      centrality = collision.centFT0C();
+      if (centAxisType == 1) {
+        centrality = collision.centFT0M();
+      } else if (centAxisType == 2) {
+        centrality = collision.multFT0M();
+      } else if (centAxisType == 3) {
+        centrality = collision.multFT0C();
+      }
+
+      // group over mcParticles
+      const auto mcTracksTablePerMcColl = mcParticles.sliceBy(mcTracksPerMcCollisionPreslice, mcColl.globalIndex());
+
+      int nRejectedPiPlus = 0;
+      int nRejectedPiMinus = 0;
+
+      int nK0s = 0;
+      int nPiPlus = 0;
+      int nPiMinus = 0;
+      int nKaPlus = 0;
+      int nKaMinus = 0;
+      int nProton = 0;
+      int nPBar = 0;
+      int nElPlus = 0;
+      int nElMinus = 0;
+      int nDePlus = 0;
+      int nDeMinus = 0;
+      int nTrack = 0;
+      int nKaon = 0;
+
+      for (const auto& mcTrack : mcTracksTablePerMcColl) {
+        if (!mcTrack.isPhysicalPrimary()) {
+          continue;
+        }
+
+        if (mcTrack.pdgCode() == 310 &&
+            0.1 < mcTrack.pt() && mcTrack.pt() < 1.5 &&
+            std::abs(mcTrack.y()) < 0.5) {
+          nK0s++;
+          fillGenTrackQA<genAnalysisDir, kK0s>(genAnalysis, mcTrack);
+        }
+
+        if (mcTrack.pt() <= 0.15 || mcTrack.pt() >= 2.0 || std::abs(mcTrack.eta()) >= 0.8) {
+          continue;
+        }
+
+        if (mcTrack.pdgCode() == 211) {
+          fillGenTrackQA<genAnalysisDir, kPi>(genAnalysis, mcTrack);
+          nPiPlus++;
+        } else if (mcTrack.pdgCode() == -211) {
+          fillGenTrackQA<genAnalysisDir, kPi>(genAnalysis, mcTrack);
+          nPiMinus++;
+        } else if (mcTrack.pdgCode() == 321) {
+          fillGenTrackQA<genAnalysisDir, kKa>(genAnalysis, mcTrack);
+          nKaPlus++;
+        } else if (mcTrack.pdgCode() == -321) {
+          fillGenTrackQA<genAnalysisDir, kKa>(genAnalysis, mcTrack);
+          nKaMinus++;
+        } else if (mcTrack.pdgCode() == 2212) {
+          fillGenTrackQA<genAnalysisDir, kPr>(genAnalysis, mcTrack);
+          nProton++;
+        } else if (mcTrack.pdgCode() == -2212) {
+          fillGenTrackQA<genAnalysisDir, kPr>(genAnalysis, mcTrack);
+          nPBar++;
+        } else if (mcTrack.pdgCode() == 11) {
+          fillGenTrackQA<genAnalysisDir, kEl>(genAnalysis, mcTrack);
+          nElPlus++;
+        } else if (mcTrack.pdgCode() == -11) {
+          fillGenTrackQA<genAnalysisDir, kEl>(genAnalysis, mcTrack);
+          nElMinus++;
+        } else if (mcTrack.pdgCode() == 1000010020) {
+          fillGenTrackQA<genAnalysisDir, kDe>(genAnalysis, mcTrack);
+          nDePlus++;
+        } else if (mcTrack.pdgCode() == -1000010020) {
+          fillGenTrackQA<genAnalysisDir, kDe>(genAnalysis, mcTrack);
+          nDeMinus++;
+        }
```

Comment:
Please, for further iterations, use named constants from `TPDGCode.h` or `PhysicsConstants.h` instead of plain numbers